Frameworks/Social/_SLFBComposeViewController.mm

```diff
@@ -0,0 +1,152 @@
+    NSString* service = [NSString stringWithFormat:@"com.facebook.sdk.tokencache.%@", [[NSBundle mainBundle] bundleIdentifier]];
+
+    NSDictionary* query = @{
+        (__bridge id)kSecClass : (__bridge id)kSecClassGenericPassword,
+        (__bridge id)kSecAttrService : service,
+        (__bridge id)kSecAttrAccount : (__bridge id)kFBSDKAccessTokenKeychainKey,
+        (__bridge id)kSecReturnData : @YES,
+        (__bridge id)kSecMatchLimit : (__bridge id)kSecMatchLimitOne
+    };
+
+    woc::unique_cf<CFTypeRef> data;
+    OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query, &data);
+    if (status != errSecSuccess) {
+        TraceError(TAG, L"Cached token not found, error = %u", (unsigned)status);
+        return nil;
+    }
+
+    if (!data) {
+        TraceError(TAG, L"Cached token not set");
+        return nil;
+    }
+
+    if (CFGetTypeID(data) != CFDataGetTypeID()) {
+        TraceError(TAG, L"Cached token unexpected type: %lu", (unsigned long)CFGetTypeID(data));
+        return nil;
+    }
+
+    NSDictionary* dict = [NSKeyedUnarchiver unarchiveObjectWithData:(__bridge NSData*)data.get()];
+
+    if (![dict isKindOfClass:[NSDictionary class]]) {
+        NSTraceError(TAG, @"Cached token data unexpected type: %@", [dict class]);
+        return nil;
+    }
+
+    // Pull the token out of the cached data we found
+    NSUserDefaults* defaults = [NSUserDefaults standardUserDefaults];
+    NSString* uuid = [defaults objectForKey:kFBSDKAccessTokenUserDefaultsKey];
+    id uuidFound = dict[kFBSDKAccessTokenUUIDKey];
+
+    if ([uuidFound isKindOfClass:[NSString class]]) {
+        if ([uuidFound isEqualToString:uuid]) {
+            id tokenData = dict[kFBSDKAccessTokenEncodedKey];
+            if ([tokenData isKindOfClass:[NSData class]]) {
+                // The access token will be a serialized form of some Facebook object.
+                // The unarchiving will fail if the object class is not registered,
+                // meaning that the Facebook SDK hasn't been loaded into this process.
+                // It would be preferable if the token were stored in some documented
+                // format so that the Social kit is usable even by callers that haven't
+                // loaded the Facebook SDK.
+                id token = [NSKeyedUnarchiver unarchiveObjectWithData:tokenData];
+                if ([token respondsToSelector:@selector(tokenString)]) {
+                    id tokenString = [token performSelector:@selector(tokenString)];
+                    if ([tokenString isKindOfClass:[NSString class]]) {
+                        return tokenString;
+                    }
+                }
+            }
+        }
+    }
+
+    return nil;
+}
+
+@implementation _SLFBComposeViewController {
+    ComPtr<ShareSheetControl> _shareSheet;
+}
+
+- (instancetype)init {
+    if (self = [super init]) {
+        NSString* token = _getAccessToken();
```

Comment:
Yes.

---
In reply to: [108001139](https://github.com/Microsoft/WinObjC/pull/2314#discussion_r108001139) [](ancestors = 108001139)